web-portal/store/partner.js

```
@@ -0,0 +1,38 @@
+import { useRoute } from 'vue-router'
+
+export const state = () => ({
+  // string | null
+  name: null,
+  // { id: string; name: string; logo_url: string } | null
+  partner: null
+})
+
+export const getters = {
+  partner: (state) => {
+    return state.partner
+  }
+}
+
+export const mutations = {
+  SET_PARTNER: (state, payload) => {
+    state.name = payload?.name ?? null
+    state.partner = payload
+  }
+}
+
+export const actions = {
+  LoadPartner: async (context) => {
+    const { query } = useRoute();
+    if (query.partner && query.partner !== context.state.name) {
+      const { data, error } = await useAsyncData('partner-fetch', () =>
+        useNuxtApp().$apiService.get(`/partners/name/${query.partner}`)
+        .catch(() => null) // eat 404s to avoid breaking everything
```

In the event of a malformed `partner` query param, we presumably want this to fail silently and display the Infinite logo as if unset.